consensus/polybft/consensus_runtime.go

```diff
@@ -1085,34 +1091,36 @@ func (c *consensusRuntime) ID() []byte {
 	return c.config.Key.Address().Bytes()
 }
 
-// MaximumFaultyNodes returns the maximum number of faulty nodes based on the validator set
-func (c *consensusRuntime) MaximumFaultyNodes() uint64 {
-	return uint64(c.fsm.validators.Len()-1) / 3
-}
-
-// Quorum returns what is the quorum size for the specified block height
-func (c *consensusRuntime) Quorum(_ uint64) uint64 {
-	return uint64(getQuorumSize(c.fsm.validators.Len()))
-}
-
 // HasQuorum returns true if quorum is reached for the given blockNumber
 func (c *consensusRuntime) HasQuorum(
 	blockNumber uint64,
 	messages []*proto.Message,
 	msgType proto.MessageType,
 ) bool {
-	quorum := c.Quorum(blockNumber)
+	// if we are not using consensus engine for current block return false
+	if c.fsm.parent.Number+1 != blockNumber {
+		c.logger.Warn("HasQuorum checking against stale block",
+			"blockNumber", blockNumber, "parentBlock", c.fsm.parent.Number)
+
+		return false
+	}
 
+	// extract the addresses of all the senders of the messages
+	senders := make([]types.Address, len(messages))
```

Comment:
Are messages unique?

Reply:
They should be, but I think we should rely on a map of senders here (instead of relying on slice).

Let me try change it that way.

Reply:
Will have another task for that validation, at the moment we are relying on go-ibft to send unique and proper messages. That is the HasQourum discussion on Slack  :)